lib/view_component/base.rb

```diff
@@ -135,6 +146,60 @@ def with(area, content = nil, &block)
       nil
     end
 
+    # Build a Slot on a component.
+    #
+    # slot: Name of Slot, in symbol form
+    # **args: Arguments to be passed to Slot initializer
+    #
+    # For example:
+    # <%= render(SlotsComponent.new) do |component| %>
+    #   <% component.slot(:footer, class_names: "footer-class") do %>
+    #     <p>This is my footer!</p>
+    #   <% end %>
+    # <% end %>
+    #
+    def slot(slot, **args, &block)
+      # Raise ArgumentError if `slot` does not exist
+      unless slot_names.include?(slot)
+        raise ArgumentError.new "Unknown slot '#{slot}' - expected one of '#{slot_names}'"
+      end
+
+      slot_class = "#{self.class.name}::#{slot.to_s.capitalize}".constantize
+
+      # Instantiate Slot class, accommodating Slots that don't accept arguments
+      slot_instance =
+        if args.present?
+          slot_class.new(args)
+        else
+          slot_class.new
+        end
+
+      # Capture block and assign to slot_instance#content
+      slot_instance.content = view_context.capture(&block) if block_given?
+
+      if slot_class.superclass == ViewComponent::CollectionSlot
```

Comment:
Should we make this `.is_a?` vs superclass? Can we imagine situations where there are multiple classes in hierarchy?